docs/guides/circuits.rst

```diff
@@ -0,0 +1,439 @@
+
+Here are examples of how to get the above information about our ``qaoa_circ``
+
+.. literalinclude:: /examples/circuits_guide.py
+  :language: python
+  :start-at: ic(qaoa_circ)
+  :end-at: ic(op.gate.name,
+
+That will produce this output:
+
+.. code-block:: text
+
+  ic| qaoa_circ: Circuit(operations=[H(0), H(1), H(2), CNOT(0,1), RZ(beta_0)(1), CNOT(0,1), CNOT(0,2), RZ(beta_1)(2), CNOT(0,2), CNOT(1,2), RZ(beta_2)(2), CNOT(1,2), RX(gamma_0)(0), RX(gamma_1)(1), RX(gamma_2)(2)], n_qubits=3)
+  ic| qaoa_circ.n_qubits: 3
+  ic| len(qaoa_circ.operations): 15
+  ic| op.gate.name: 'H', op.qubit_indices: (0,), op.params: ()
+  ic| op.gate.name: 'H', op.qubit_indices: (1,), op.params: ()
+  ic| op.gate.name: 'H', op.qubit_indices: (2,), op.params: ()
+  ic| op.gate.name: 'CNOT', op.qubit_indices: (0, 1), op.params: ()
+  ic| op.gate.name: 'RZ', op.qubit_indices: (1,), op.params: (beta_0,)
+  ic| op.gate.name: 'CNOT', op.qubit_indices: (0, 1), op.params: ()
+  ic| op.gate.name: 'CNOT', op.qubit_indices: (0, 2), op.params: ()
+  ic| op.gate.name: 'RZ', op.qubit_indices: (2,), op.params: (beta_1,)
+  ic| op.gate.name: 'CNOT', op.qubit_indices: (0, 2), op.params: ()
+  ic| op.gate.name: 'CNOT', op.qubit_indices: (1, 2), op.params: ()
+  ic| op.gate.name: 'RZ', op.qubit_indices: (2,), op.params: (beta_2,)
+  ic| op.gate.name: 'CNOT', op.qubit_indices: (1, 2), op.params: ()
+  ic| op.gate.name: 'RX', op.qubit_indices: (0,), op.params: (gamma_0,)
+  ic| op.gate.name: 'RX', op.qubit_indices: (1,), op.params: (gamma_1,)
+  ic| op.gate.name: 'RX', op.qubit_indices: (2,), op.params: (gamma_2,)
+
+Powers of gates and gate exponentials
+-------------------------------------
+
+You can add a power of arbitrary non-symbolic gates to your circuit, as well as get the gate exponential. This won't be used in the creation of our QAOA circuit, but we'll still provide examples here. 
+
+**To raise a gate to a power**, you can use ``.power(exponent)`` with the ``exponent`` argument the power you want to raise the gate to. Here we verify that the Hadamard gate raised to the 2nd power is the same as the identity.
+
+.. literalinclude:: /examples/circuits_guide.py
+  :language: python
+  :start-at: ic(H.power(2).matrix == sympy.eye(2))
+  :end-at: ic(H.power(2).matrix == sympy.eye(2))
+
+Where the output is ``ic| H.power(2).matrix == sympy.eye(2): True``
+
+**To get the gate exponential** of a specific gate, you can use ``.exp``, shown in this example with ``.matrix`` to view what the matrix looks like. It is also generally a good idea to simplify the matrix when possible.
+
+.. literalinclude:: /examples/circuits_guide.py
+  :language: python
+  :start-at: ic(sympy.simplify(H.exp.matrix))
+  :end-at: ic(sympy.simplify(H.exp.matrix))
+
+
+.. _wavefunction_operations:
+
+Gate operations vs Wave Function Operations
```

Comment:
I believe Wavefunction is one word.

```suggestion
Gate operations vs Wavefunction Operations
```

Reply:
Hmmm not according to [Wikipedia](https://en.wikipedia.org/wiki/Wave_function), [Britannica](https://www.britannica.com/science/wave-function), and this [openstax textbook](https://openstax.org/books/university-physics-volume-3/pages/7-1-wave-functions). Unless this is our own standard to have it all one word, I think best to leave it 😄

Reply:
Ok, fair point! 
Then at least to be consistent with upper-casing please do "Gate operations vs wave function operations"?

Reply:
Also fair point! Will change 👍

Reply:
fixed in 9b6a11e3894cfc2026f441d43179d599a8291a33